### PR TITLE
Fix BigVM xena issues

### PR DESCRIPTION
**Pull request by grandchild**
- bigvm: Fix ResourceRequest._add_resource() usage
- bigvm/ComputeNodeList: Only get non-deleted nodes
- bigvm: Use public SchedulerReportClient.get_provider_traits
- bigvm: Catch possible KeyError from get_inventory
- bigvm: Replace removed _update_inventory() call

### CODE REVIEW

File: nova/bigvm/manager.py [hunk 1]
```diff
@@ -403,7 +403,7 @@ def _get_providers(self, context):
         # retrieve all bigvm provider's inventories
         for rp_uuid, rp in bigvm_providers.items():
             inventory = client._get_inventory(context, rp_uuid)
-            rp['inventory'] = inventory['inventories']
```
Comment (joker-at-work) [inline nova/bigvm/manager.py:406]: Does the later code handle `inventory` properly? From a quick check, it looks to me like the code assumes `rp['inventory']` to always be a dictionary.

**Review by joker-at-work — COMMENTED** [review]
The last commit looks ok to me, I'm just wondering why we don't catch all exceptions, just the ones raised explicitly. That's afaics a change to how it worked previously.

**Comment by grandchild** [comment]
Re exceptions: Instinctively I'd like to catch only expected exceptions and not random Key-/AtrributeErrors from somewhere else. Those I'd like to see as crashes/errors.

**Comment by joker-at-work** [comment]
> Re exceptions: Instinctively I'd like to catch only expected exceptions and not random Key-/AtrributeErrors from somewhere else. Those I'd like to see as crashes/errors.

I get that point, but we already created the resource-provider at this point and I think it would be good to at least log some helpful information (probably `rp['host']`) and re-raise, so we have the information around when looking trying to figure out why a certain host isn't getting used a bigvm host.

There is another problem with the "die on exception"-approach, I can see: While we will be scheduled to run again, we might run into the same issue again. This prohibits other hosts, that would be visited by the for-loop after the breaking host, from getting their inventories set/freed up completely. Therefore, I'd still catch all exceptions, I think, instead of "crashing" the function call - not the whole agent.

**Review by joker-at-work — APPROVED** [review]

**Pull request merged**